File: data-structures/submissions/a1/Hidden.java
public class Hidden {
    public static boolean isHidden(String s, String t) {
        String wordSoFar = "";
        int scan = 0;
        for (int i = 0; i < s.length();i++) {
            if (s.charAt(i) == t.charAt(scan)) {
                wordSoFar = wordSoFar.concat(String.valueOf(s.charAt(i)));
                scan++;
            }
            if (wordSoFar.equals(t)) {
                return true;
            }
        }
        return false;
    }

    public static void main(String[] args) {
        System.out.println(Boolean.toString(isHidden("welcometothehotelcalifornia","melon")));
        System.out.println(Boolean.toString(isHidden("welcometothehotelcalifornia","space")));
        System.out.println(Boolean.toString(isHidden("TQ89MnQU3IC7t6","MUIC")));
        System.out.println(Boolean.toString(isHidden("VhHTdipc07","htc")));
        System.out.println(Boolean.toString(isHidden("VhHTdipc07","hTc")));
    }
}

File: intro-to-programming/Assignment 5/hidden.py
# Assignment 5, Task 2
# Name: Theeradon Sarawek
# Collaborators:
# Time Spent: 15 minutes


def is_hidden(s:str ,t:str) -> bool:
        s = s.lower() #Ignore cases
        t = t.lower() #Ignore cases

        foundIndex = 0
        for char in s:
            if foundIndex == len(t): #Exceeds the index, meaning that all characters have already been found in the order.
                return True
            if char == t[foundIndex]:
                foundIndex += 1
        return False

def test_isHidden():
    assert is_hidden("welcometothehotelcalifornia", "melon") == True
    assert is_hidden("welcometothehotelcalifornia", "space") == False
    assert is_hidden("TQ89MnQU3IC7t6", "MUIC") == True
    assert is_hidden("VhHTdipc07", "htc") == True
    assert is_hidden("VhHTdipc07", "hTc") == True
    assert is_hidden("VhHTdipc07", "vat") == False

test_isHidden() #Passed the assertion test
